cmd/npmExecuteLint.go

```diff
@@ -0,0 +1,150 @@
+package cmd
+
+import (
+	"encoding/json"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/npm"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"path"
+	"strconv"
+	"strings"
+)
+
+func npmExecuteLint(config npmExecuteLintOptions, telemetryData *telemetry.CustomData) {
+	utils := npm.NpmUtilsBundle{}
+
+	err := runNpmExecuteLint(&utils)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runNpmExecuteLint(utils npm.NpmUtils) error {
+	packageJSONFiles, err := npm.FindPackageJSONFiles(utils)
+	if err != nil {
+		return err
+	}
+
+	packagesWithCilint, err := findPackagesWithCilint(packageJSONFiles, utils)
+	if err != nil {
+		return err
+	}
+
+	if len(packagesWithCilint) > 0 {
+		err = runCiLint(packagesWithCilint, utils)
+		if err != nil {
+			return err
+		}
+	} else {
+		err = runDefaultLint(utils)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func findPackagesWithCilint(packageJSONFiles []string, utils npm.NpmUtils) ([]string, error) {
+	var packageJSON map[string]interface{}
+	var packagesWithCilint []string
+
+	for _, file := range packageJSONFiles {
+		packageRaw, err := utils.FileRead(file)
+		if err != nil {
+			return nil, err
+		}
+
+		json.Unmarshal(packageRaw, &packageJSON)
+
+		scripts, ok := packageJSON["scripts"].(map[string]interface{})
+		if ok {
+			_, ok := scripts["ci-lint"].(string)
+			if ok {
+				packagesWithCilint = append(packagesWithCilint, file)
+				log.Entry().Info("Discovered ci-lint script in " + file)
+			}
+		}
+		packageJSON = nil
+	}
+	return packagesWithCilint, nil
+}
+
+func runCiLint(packagesWithCilint []string, utils npm.NpmUtils) error {
+	execRunner := utils.GetExecRunner()
+	oldWorkingDirectory, err := utils.Getwd()
+	if err != nil {
+		return err
+	}
+
+	for _, packageJSON := range packagesWithCilint {
+		dir := path.Dir(packageJSON)
+		err := utils.Chdir(dir)
+		if err != nil {
+			return err
```

Comment:
I would wrap the error here, since the context will be missing completely in the log:
```suggestion
			return fmt.Errorf("failed to change into directory for executing ci-lint script: %w", err)
```